postgres/upstream.sls

```diff
@@ -4,11 +4,17 @@
 {%- if 'pkg_repo' in postgres -%}
 
   {%- if postgres.use_upstream_repo -%}
-
 # Add upstream repository for your distro
+
 postgresql-repo:
   pkgrepo.managed:
     {{- format_kwargs(postgres.pkg_repo) }}
+    {% if grains.os_family == 'Suse' %}
+  cmd.run:
+    - name: zypper --gpg-auto-import-keys refresh --force
```

Comment:
This is in connection with what we've discussed about requisites for the `pkg` state.
I know it is not obvious from the Salt docs, but `pkgrepo.managed` state supports special option `gpgautoimport` when target package manager is Zypper.
No need for messing with grains here. I'll show how simple it could be done above at `osmap.yaml` file.

Reply:
How did you know that ;-)   I did check the [docs](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkgrepo.html).